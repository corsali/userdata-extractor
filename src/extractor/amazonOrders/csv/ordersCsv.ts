import config from "../../../config/index.js";
import { CsvExtractor } from "../../csvExtractor.js";
import { Orders } from "../models/orders.js";

const extractPrice = (price: string) =>
  parseFloat(/[\d,]*\.[\d]*/.exec(price)?.[0]);

class OrdersCsv extends CsvExtractor {
  async process() {
    await this.parse();

    if (this.csvDocument) {
      const mappedRow = this.csvDocument.map(
        (order: any) =>
          new Orders({
            orderDate: order["order date"],
            orderId: order["order id"],
            title: order.title,
            category: order.category,
            asinIsbn: order["asin/isbn"],
            unspscCode: order["unspsc code"],
            website: order.website,
            releaseDate: order["release date"],
            condition: order.condition,
            seller: order.seller,
            sellerCredentials: order["seller credentials"],
            listPricePerUnit: extractPrice(order["list price per unit"]),
            purchasePricePerUnit: extractPrice(
              order["purchase price per unit"]
            ),
            quantity: order.quantity,
            paymentInstrumentType: order["payment instrument type"],
            purchaseOrderNumber: order["purchase order number"],
            poLineNumber: order["po line number"],
            orderingCustomerEmail: order["ordering customer email"],
            shipmentDate: order["shipment date"],
            shippingAddressName: order["shipping address name"],
            shippingAddressStreet1: order["shipping address street 1"],
            shippingAddressStreet2: order["shipping address street 2"],
            shippingAddressCity: order["shipping address city"],
            shippingAddressState: order["shipping address state"],
            shippingAddressZip: order["shipping address zip"],
            orderStatus: order["order status"],
            carrierNameAndTrackingNumber:
              order["carrier name & tracking number"],
            itemSubtotal: extractPrice(order["item subtotal"]),
            itemSubtotalTax: extractPrice(order["item subtotal tax"]),
            itemTotal: extractPrice(order["item total"]),
            taxExemptionApplied: order["tax exemption applied"],
            taxExemptionType: order["tax exemption type"],
            exemptionOptOut: order["exemption opt-out"],
            buyerName: order["buyer name"],
            currency: order.currency,
            groupName: order["group name"],
          })
      );

      this.table.rows.push(...mappedRow);
    }
  }
}

export const ordersCsv = new OrdersCsv(
  config.SERVICE_AMAZON_ORDERS,
  ".*\\.csv",
  "orders"
);
