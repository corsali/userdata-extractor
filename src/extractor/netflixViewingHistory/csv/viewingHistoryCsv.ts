import config from "../../../config/index.js";
import { CsvExtractor } from "../../csvExtractor.js";
import { ViewingHistory } from "../models/viewingHistory.js";

class ViewingHistoryCsv extends CsvExtractor {
  async process() {
    await this.parse();

    const mappedRow = this.csvDocument.map(
      (row: any) => new ViewingHistory(row.title, row.date)
    );

    this.table.rows.push(...mappedRow);
  }
}

export const viewingHistoryCsv = new ViewingHistoryCsv(
  config.SERVICE_NETFLIX_VIEWING_HISTORY,
  ".*NetflixViewingHistory.csv",
  "viewing_history"
);
