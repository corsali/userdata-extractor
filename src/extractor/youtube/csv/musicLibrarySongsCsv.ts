import config from "../../../config/index.js";
import { CsvExtractor } from "../../csvExtractor.js";
import { MusicLibrarySongs } from "../models/musicLibrarySongs.js";

class MusicLibrarySongsCsv extends CsvExtractor {
  async process() {
    this.parse();

    const mappedRow = this.csvDocument.map(
      (row: any) =>
        new MusicLibrarySongs({
          songUrl: row["song url"],
          songTitle: row["song title"],
          albumTitle: row["album title"],
          artistNames: row["artist names"],
        })
    );

    this.table.rows.push(...mappedRow);
  }
}

export const musicLibrarySongsCsv = new MusicLibrarySongsCsv(
  config.SERVICE_YOUTUBE,
  ".*/music-library-songs.csv",
  "music_library_songs"
);
