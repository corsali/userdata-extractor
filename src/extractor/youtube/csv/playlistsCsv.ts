import config from "../../../config/index.js";
import { CsvExtractor } from "../../csvExtractor.js";
import { Playlists } from "../models/playlists.js";

class PlaylistsCsv extends CsvExtractor {
  async process() {
    const playlistHeader = await this.parseText(
      this.fileContents.split("\n").slice(0, 2).join("\n")
    );

    const playlistItems = await this.parseText(
      this.fileContents.split("\n").slice(2).join("\n")
    );

    const mappedRow = playlistItems.map(
      (playlistItem: any) =>
        new Playlists({
          playlistId: playlistHeader[0]["playlist id"],
          channelId: playlistHeader[0]["channel id"],
          dateCreated: playlistHeader[0]["time created"],
          dateUpdated: playlistHeader[0]["time updated"],
          playlistTitle: playlistHeader[0].title,
          playlistDescription: playlistHeader[0].description,
          playlistVisibility: playlistHeader[0].visibility,
          itemVideoId: playlistItem["video id"],
          dateItemVideoAdded: playlistItem["time added"],
        })
    );

    this.table.rows.push(...mappedRow);
  }
}

export const playlistsCsv = new PlaylistsCsv(
  config.SERVICE_YOUTUBE,
  "playlists/[^/]*\\.csv",
  "playlists"
);
