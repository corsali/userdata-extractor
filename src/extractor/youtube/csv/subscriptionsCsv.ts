import config from "../../../config/index.js";
import { CsvExtractor } from "../../csvExtractor.js";
import { Subscriptions } from "../models/subscriptions.js";

class SubscriptionsCsv extends CsvExtractor {
  async process() {
    this.parse({ skipEmptyLines: true });

    const mappedRow = this.csvDocument.map(
      (row: any) =>
        new Subscriptions(
          row["Channel Id"],
          row["Channel Url"],
          row["Channel Title"]
        )
    );

    this.table.rows.push(...mappedRow);
  }
}

export const subscriptionsCsv = new SubscriptionsCsv(
  config.SERVICE_YOUTUBE,
  ".*/subscriptions.csv",
  "subscriptions"
);
