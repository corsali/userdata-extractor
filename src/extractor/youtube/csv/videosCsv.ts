import config from "../../../config/index.js";
import { CsvExtractor } from "../../csvExtractor.js";
import { Videos } from "../models/videos.js";

class VideosCsv extends CsvExtractor {
  async process() {
    await this.parse();

    const mappedRow = this.csvDocument.map(
      (row: any) =>
        new Videos({
          videoId: row["video id"],
          channelId: row["channel id"],
          title: row.title,
          status: row.status,
          visibility: row.visibility,
          timeCreated: row["time created"],
          timePublished: row["time published"],
          duration: row.duration,
          description: row.description,
          category: row.category,
          locationAddress: row["location address"],
          locationCountry: row["location country"],
          timeRecorded: row["time recorded"],
          likeCount: row["like count"],
          dislikeCount: row["dislike count"],
          viewCount: row["view count"],
          syndication: row.syndication,
          license: row.license,
          embeddable: row.embeddable,
        })
    );

    this.table.rows.push(...mappedRow);
  }
}

export const videosCsv = new VideosCsv(
  config.SERVICE_YOUTUBE,
  ".*/video metadata.csv",
  "videos"
);
