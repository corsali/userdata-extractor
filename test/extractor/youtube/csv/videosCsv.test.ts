import { videosCsv } from "../../../../src/extractor/youtube/csv/videosCsv";
import { loadTestFileAsText } from "../../../helper";

describe("Videos (CSV)", () => {
  test("it should load file correctly", async () => {
    const data = await loadTestFileAsText(
      "/csv/youtube/Takeout/YouTube and YouTube Music/videos/Video Metadata.csv"
    );

    videosCsv.fileContents = data;

    await videosCsv.process();
    const { rows } = videosCsv.table;

    expect(rows).toMatchSnapshot();
  });
});
