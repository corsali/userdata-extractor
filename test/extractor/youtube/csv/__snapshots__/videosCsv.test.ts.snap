// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Videos (CSV) it should load file correctly 1`] = `
Array [
  Videos {
    "category": TextTableValue {
      "type": "text",
      "value": "News & Politics",
    },
    "channel_id": TextTableValue {
      "type": "text",
      "value": "UCoY-bu7ZpUhCOd2sJpLGQ2w",
    },
    "description": TextTableValue {
      "type": "text",
      "value": "OFallon Missouri Walmart. This story aired on FOX 2 KTVI in 2004",
    },
    "dislike_count": IntegerTableValue {
      "type": "integer",
      "value": 0,
    },
    "duration": TextTableValue {
      "type": "text",
      "value": "138 seconds",
    },
    "embeddable": TextTableValue {
      "type": "text",
      "value": "allowed",
    },
    "license": TextTableValue {
      "type": "text",
      "value": "standard YouTube",
    },
    "like_count": IntegerTableValue {
      "type": "integer",
      "value": 3,
    },
    "location_address": TextTableValue {
      "type": "text",
      "value": "Walmart Supercenter",
    },
    "location_country": TextTableValue {
      "type": "text",
      "value": "",
    },
    "status": TextTableValue {
      "type": "text",
      "value": "processed",
    },
    "syndication": TextTableValue {
      "type": "text",
      "value": "all platforms",
    },
    "time_created": DateTableValue {
      "type": "date",
      "value": 1360726900000,
    },
    "time_published": DateTableValue {
      "type": "date",
      "value": 1360732836000,
    },
    "time_recorded": TextTableValue {
      "type": "text",
      "value": "",
    },
    "title": TextTableValue {
      "type": "text",
      "value": "I was interviewed by FOX 2 KTVI St. Louis.",
    },
    "video_id": TextTableValue {
      "type": "text",
      "value": "5ocjEDUyC3E",
    },
    "view_count": IntegerTableValue {
      "type": "integer",
      "value": 325,
    },
    "visibility": TextTableValue {
      "type": "text",
      "value": "Public",
    },
  },
  Videos {
    "category": TextTableValue {
      "type": "text",
      "value": "People & Blogs",
    },
    "channel_id": TextTableValue {
      "type": "text",
      "value": "UCoY-bu7ZpUhCOd2sJpLGQ2w",
    },
    "description": TextTableValue {
      "type": "text",
      "value": "Ethan dancing, this will be good to show his first girlfriend!",
    },
    "dislike_count": IntegerTableValue {
      "type": "integer",
      "value": 0,
    },
    "duration": TextTableValue {
      "type": "text",
      "value": "33 seconds",
    },
    "embeddable": TextTableValue {
      "type": "text",
      "value": "allowed",
    },
    "license": TextTableValue {
      "type": "text",
      "value": "standard YouTube",
    },
    "like_count": IntegerTableValue {
      "type": "integer",
      "value": 2,
    },
    "location_address": TextTableValue {
      "type": "text",
      "value": "",
    },
    "location_country": TextTableValue {
      "type": "text",
      "value": "",
    },
    "status": TextTableValue {
      "type": "text",
      "value": "processed",
    },
    "syndication": TextTableValue {
      "type": "text",
      "value": "all platforms",
    },
    "time_created": DateTableValue {
      "type": "date",
      "value": 1423235417000,
    },
    "time_published": DateTableValue {
      "type": "date",
      "value": 1423235743000,
    },
    "time_recorded": TextTableValue {
      "type": "text",
      "value": "",
    },
    "title": TextTableValue {
      "type": "text",
      "value": "Ethan dancing",
    },
    "video_id": TextTableValue {
      "type": "text",
      "value": "JavYYZ960-s",
    },
    "view_count": IntegerTableValue {
      "type": "integer",
      "value": 26,
    },
    "visibility": TextTableValue {
      "type": "text",
      "value": "Public",
    },
  },
  Videos {
    "category": TextTableValue {
      "type": "text",
      "value": "People & Blogs",
    },
    "channel_id": TextTableValue {
      "type": "text",
      "value": "UCoY-bu7ZpUhCOd2sJpLGQ2w",
    },
    "description": TextTableValue {
      "type": "text",
      "value": "Please note that I have no type of video editing knowledge. This footage is unedited. This mall is located in Chesterfield, Missouri",
    },
    "dislike_count": IntegerTableValue {
      "type": "integer",
      "value": 0,
    },
    "duration": TextTableValue {
      "type": "text",
      "value": "543 seconds",
    },
    "embeddable": TextTableValue {
      "type": "text",
      "value": "allowed",
    },
    "license": TextTableValue {
      "type": "text",
      "value": "standard YouTube",
    },
    "like_count": IntegerTableValue {
      "type": "integer",
      "value": 16,
    },
    "location_address": TextTableValue {
      "type": "text",
      "value": "Chesterfield Mall",
    },
    "location_country": TextTableValue {
      "type": "text",
      "value": "",
    },
    "status": TextTableValue {
      "type": "text",
      "value": "processed",
    },
    "syndication": TextTableValue {
      "type": "text",
      "value": "all platforms",
    },
    "time_created": DateTableValue {
      "type": "date",
      "value": 1568472059000,
    },
    "time_published": DateTableValue {
      "type": "date",
      "value": 1568472996000,
    },
    "time_recorded": TextTableValue {
      "type": "text",
      "value": "",
    },
    "title": TextTableValue {
      "type": "text",
      "value": "Video tour of nearly vacant Chesterfield Mall. Chesterfield, Missouri DEAD MALL September 2019.",
    },
    "video_id": TextTableValue {
      "type": "text",
      "value": "SiNkkdrjJ6Q",
    },
    "view_count": IntegerTableValue {
      "type": "integer",
      "value": 1180,
    },
    "visibility": TextTableValue {
      "type": "text",
      "value": "Public",
    },
  },
  Videos {
    "category": TextTableValue {
      "type": "text",
      "value": "People & Blogs",
    },
    "channel_id": TextTableValue {
      "type": "text",
      "value": "UCoY-bu7ZpUhCOd2sJpLGQ2w",
    },
    "description": TextTableValue {
      "type": "text",
      "value": "Last Sears in the St. Louis area is closing. It was very busy on the Sunday afternoon I was there.",
    },
    "dislike_count": IntegerTableValue {
      "type": "integer",
      "value": 0,
    },
    "duration": TextTableValue {
      "type": "text",
      "value": "585 seconds",
    },
    "embeddable": TextTableValue {
      "type": "text",
      "value": "allowed",
    },
    "license": TextTableValue {
      "type": "text",
      "value": "standard YouTube",
    },
    "like_count": IntegerTableValue {
      "type": "integer",
      "value": 2,
    },
    "location_address": TextTableValue {
      "type": "text",
      "value": "Mid Rivers Mall",
    },
    "location_country": TextTableValue {
      "type": "text",
      "value": "",
    },
    "status": TextTableValue {
      "type": "text",
      "value": "processed",
    },
    "syndication": TextTableValue {
      "type": "text",
      "value": "all platforms",
    },
    "time_created": DateTableValue {
      "type": "date",
      "value": 1569788892000,
    },
    "time_published": DateTableValue {
      "type": "date",
      "value": 1569789380000,
    },
    "time_recorded": TextTableValue {
      "type": "text",
      "value": "",
    },
    "title": TextTableValue {
      "type": "text",
      "value": "Video tour of a closing Sears store at Mid Rivers Mall September 2019",
    },
    "video_id": TextTableValue {
      "type": "text",
      "value": "rUmIEpGmY5I",
    },
    "view_count": IntegerTableValue {
      "type": "integer",
      "value": 172,
    },
    "visibility": TextTableValue {
      "type": "text",
      "value": "Public",
    },
  },
  Videos {
    "category": TextTableValue {
      "type": "text",
      "value": "People & Blogs",
    },
    "channel_id": TextTableValue {
      "type": "text",
      "value": "UCoY-bu7ZpUhCOd2sJpLGQ2w",
    },
    "description": TextTableValue {
      "type": "text",
      "value": "",
    },
    "dislike_count": IntegerTableValue {
      "type": "integer",
      "value": 0,
    },
    "duration": TextTableValue {
      "type": "text",
      "value": "104 seconds",
    },
    "embeddable": TextTableValue {
      "type": "text",
      "value": "allowed",
    },
    "license": TextTableValue {
      "type": "text",
      "value": "standard YouTube",
    },
    "like_count": IntegerTableValue {
      "type": "integer",
      "value": 0,
    },
    "location_address": TextTableValue {
      "type": "text",
      "value": "St Peters Condominium Association",
    },
    "location_country": TextTableValue {
      "type": "text",
      "value": "",
    },
    "status": TextTableValue {
      "type": "text",
      "value": "processed",
    },
    "syndication": TextTableValue {
      "type": "text",
      "value": "all platforms",
    },
    "time_created": DateTableValue {
      "type": "date",
      "value": 1575397717000,
    },
    "time_published": DateTableValue {
      "type": "date",
      "value": null,
    },
    "time_recorded": TextTableValue {
      "type": "text",
      "value": "",
    },
    "title": TextTableValue {
      "type": "text",
      "value": "784 Del Ray Dr",
    },
    "video_id": TextTableValue {
      "type": "text",
      "value": "LIJoNu2-Uew",
    },
    "view_count": IntegerTableValue {
      "type": "integer",
      "value": 25,
    },
    "visibility": TextTableValue {
      "type": "text",
      "value": "Unlisted",
    },
  },
  Videos {
    "category": TextTableValue {
      "type": "text",
      "value": "People & Blogs",
    },
    "channel_id": TextTableValue {
      "type": "text",
      "value": "UCoY-bu7ZpUhCOd2sJpLGQ2w",
    },
    "description": TextTableValue {
      "type": "text",
      "value": "Great job Ethan!",
    },
    "dislike_count": IntegerTableValue {
      "type": "integer",
      "value": 0,
    },
    "duration": TextTableValue {
      "type": "text",
      "value": "24 seconds",
    },
    "embeddable": TextTableValue {
      "type": "text",
      "value": "allowed",
    },
    "license": TextTableValue {
      "type": "text",
      "value": "standard YouTube",
    },
    "like_count": IntegerTableValue {
      "type": "integer",
      "value": 1,
    },
    "location_address": TextTableValue {
      "type": "text",
      "value": "Philadelphia",
    },
    "location_country": TextTableValue {
      "type": "text",
      "value": "",
    },
    "status": TextTableValue {
      "type": "text",
      "value": "processed",
    },
    "syndication": TextTableValue {
      "type": "text",
      "value": "all platforms",
    },
    "time_created": DateTableValue {
      "type": "date",
      "value": 1595100635000,
    },
    "time_published": DateTableValue {
      "type": "date",
      "value": 1595101161000,
    },
    "time_recorded": TextTableValue {
      "type": "text",
      "value": "",
    },
    "title": TextTableValue {
      "type": "text",
      "value": "Ethan learning to drive",
    },
    "video_id": TextTableValue {
      "type": "text",
      "value": "JNGdUh7hwRw",
    },
    "view_count": IntegerTableValue {
      "type": "integer",
      "value": 24,
    },
    "visibility": TextTableValue {
      "type": "text",
      "value": "Public",
    },
  },
  Videos {
    "category": TextTableValue {
      "type": "text",
      "value": "People & Blogs",
    },
    "channel_id": TextTableValue {
      "type": "text",
      "value": "UCoY-bu7ZpUhCOd2sJpLGQ2w",
    },
    "description": TextTableValue {
      "type": "text",
      "value": "",
    },
    "dislike_count": IntegerTableValue {
      "type": "integer",
      "value": 0,
    },
    "duration": TextTableValue {
      "type": "text",
      "value": "53 seconds",
    },
    "embeddable": TextTableValue {
      "type": "text",
      "value": "allowed",
    },
    "license": TextTableValue {
      "type": "text",
      "value": "standard YouTube",
    },
    "like_count": IntegerTableValue {
      "type": "integer",
      "value": 0,
    },
    "location_address": TextTableValue {
      "type": "text",
      "value": "",
    },
    "location_country": TextTableValue {
      "type": "text",
      "value": "",
    },
    "status": TextTableValue {
      "type": "text",
      "value": "processed",
    },
    "syndication": TextTableValue {
      "type": "text",
      "value": "all platforms",
    },
    "time_created": DateTableValue {
      "type": "date",
      "value": 1605913504000,
    },
    "time_published": DateTableValue {
      "type": "date",
      "value": null,
    },
    "time_recorded": TextTableValue {
      "type": "text",
      "value": "",
    },
    "title": TextTableValue {
      "type": "text",
      "value": "November 20, 2020",
    },
    "video_id": TextTableValue {
      "type": "text",
      "value": "38kyNPbDdGw",
    },
    "view_count": IntegerTableValue {
      "type": "integer",
      "value": 1,
    },
    "visibility": TextTableValue {
      "type": "text",
      "value": "Unlisted",
    },
  },
  Videos {
    "category": TextTableValue {
      "type": "text",
      "value": "People & Blogs",
    },
    "channel_id": TextTableValue {
      "type": "text",
      "value": "UCoY-bu7ZpUhCOd2sJpLGQ2w",
    },
    "description": TextTableValue {
      "type": "text",
      "value": "",
    },
    "dislike_count": IntegerTableValue {
      "type": "integer",
      "value": 0,
    },
    "duration": TextTableValue {
      "type": "text",
      "value": "114 seconds",
    },
    "embeddable": TextTableValue {
      "type": "text",
      "value": "allowed",
    },
    "license": TextTableValue {
      "type": "text",
      "value": "standard YouTube",
    },
    "like_count": IntegerTableValue {
      "type": "integer",
      "value": 0,
    },
    "location_address": TextTableValue {
      "type": "text",
      "value": "St Peters Condominium Association",
    },
    "location_country": TextTableValue {
      "type": "text",
      "value": "",
    },
    "status": TextTableValue {
      "type": "text",
      "value": "processed",
    },
    "syndication": TextTableValue {
      "type": "text",
      "value": "all platforms",
    },
    "time_created": DateTableValue {
      "type": "date",
      "value": 1622772818000,
    },
    "time_published": DateTableValue {
      "type": "date",
      "value": null,
    },
    "time_recorded": TextTableValue {
      "type": "text",
      "value": "",
    },
    "title": TextTableValue {
      "type": "text",
      "value": "789 Boca Raton Dr",
    },
    "video_id": TextTableValue {
      "type": "text",
      "value": "fRnpHyup9gE",
    },
    "view_count": IntegerTableValue {
      "type": "integer",
      "value": 7,
    },
    "visibility": TextTableValue {
      "type": "text",
      "value": "Unlisted",
    },
  },
  Videos {
    "category": TextTableValue {
      "type": "text",
      "value": "People & Blogs",
    },
    "channel_id": TextTableValue {
      "type": "text",
      "value": "UCoY-bu7ZpUhCOd2sJpLGQ2w",
    },
    "description": TextTableValue {
      "type": "text",
      "value": "",
    },
    "dislike_count": IntegerTableValue {
      "type": "integer",
      "value": 0,
    },
    "duration": TextTableValue {
      "type": "text",
      "value": "51 seconds",
    },
    "embeddable": TextTableValue {
      "type": "text",
      "value": "allowed",
    },
    "license": TextTableValue {
      "type": "text",
      "value": "standard YouTube",
    },
    "like_count": IntegerTableValue {
      "type": "integer",
      "value": 0,
    },
    "location_address": TextTableValue {
      "type": "text",
      "value": "",
    },
    "location_country": TextTableValue {
      "type": "text",
      "value": "",
    },
    "status": TextTableValue {
      "type": "text",
      "value": "processed",
    },
    "syndication": TextTableValue {
      "type": "text",
      "value": "all platforms",
    },
    "time_created": DateTableValue {
      "type": "date",
      "value": 1623970545000,
    },
    "time_published": DateTableValue {
      "type": "date",
      "value": null,
    },
    "time_recorded": TextTableValue {
      "type": "text",
      "value": "",
    },
    "title": TextTableValue {
      "type": "text",
      "value": "2278 Highland Hill Dr",
    },
    "video_id": TextTableValue {
      "type": "text",
      "value": "6u0XuXI8J7M",
    },
    "view_count": IntegerTableValue {
      "type": "integer",
      "value": 6,
    },
    "visibility": TextTableValue {
      "type": "text",
      "value": "Unlisted",
    },
  },
  Videos {
    "category": TextTableValue {
      "type": "text",
      "value": "People & Blogs",
    },
    "channel_id": TextTableValue {
      "type": "text",
      "value": "UCoY-bu7ZpUhCOd2sJpLGQ2w",
    },
    "description": TextTableValue {
      "type": "text",
      "value": "",
    },
    "dislike_count": IntegerTableValue {
      "type": "integer",
      "value": 0,
    },
    "duration": TextTableValue {
      "type": "text",
      "value": "31 seconds",
    },
    "embeddable": TextTableValue {
      "type": "text",
      "value": "allowed",
    },
    "license": TextTableValue {
      "type": "text",
      "value": "standard YouTube",
    },
    "like_count": IntegerTableValue {
      "type": "integer",
      "value": 0,
    },
    "location_address": TextTableValue {
      "type": "text",
      "value": "",
    },
    "location_country": TextTableValue {
      "type": "text",
      "value": "",
    },
    "status": TextTableValue {
      "type": "text",
      "value": "processed",
    },
    "syndication": TextTableValue {
      "type": "text",
      "value": "all platforms",
    },
    "time_created": DateTableValue {
      "type": "date",
      "value": 1625188452000,
    },
    "time_published": DateTableValue {
      "type": "date",
      "value": null,
    },
    "time_recorded": TextTableValue {
      "type": "text",
      "value": "",
    },
    "title": TextTableValue {
      "type": "text",
      "value": "July 1, 2021",
    },
    "video_id": TextTableValue {
      "type": "text",
      "value": "UTKqRmxYecQ",
    },
    "view_count": IntegerTableValue {
      "type": "integer",
      "value": 4,
    },
    "visibility": TextTableValue {
      "type": "text",
      "value": "Unlisted",
    },
  },
  Videos {
    "category": TextTableValue {
      "type": "text",
      "value": "People & Blogs",
    },
    "channel_id": TextTableValue {
      "type": "text",
      "value": "UCoY-bu7ZpUhCOd2sJpLGQ2w",
    },
    "description": TextTableValue {
      "type": "text",
      "value": "789 Boca Raton Dr",
    },
    "dislike_count": IntegerTableValue {
      "type": "integer",
      "value": 0,
    },
    "duration": TextTableValue {
      "type": "text",
      "value": "268 seconds",
    },
    "embeddable": TextTableValue {
      "type": "text",
      "value": "allowed",
    },
    "license": TextTableValue {
      "type": "text",
      "value": "standard YouTube",
    },
    "like_count": IntegerTableValue {
      "type": "integer",
      "value": 0,
    },
    "location_address": TextTableValue {
      "type": "text",
      "value": "St Peters Condominium Association",
    },
    "location_country": TextTableValue {
      "type": "text",
      "value": "",
    },
    "status": TextTableValue {
      "type": "text",
      "value": "processed",
    },
    "syndication": TextTableValue {
      "type": "text",
      "value": "all platforms",
    },
    "time_created": DateTableValue {
      "type": "date",
      "value": 1626052223000,
    },
    "time_published": DateTableValue {
      "type": "date",
      "value": null,
    },
    "time_recorded": TextTableValue {
      "type": "text",
      "value": "",
    },
    "title": TextTableValue {
      "type": "text",
      "value": "July 11, 2021",
    },
    "video_id": TextTableValue {
      "type": "text",
      "value": "hqHOZ0woPX0",
    },
    "view_count": IntegerTableValue {
      "type": "integer",
      "value": 11,
    },
    "visibility": TextTableValue {
      "type": "text",
      "value": "Unlisted",
    },
  },
  Videos {
    "category": TextTableValue {
      "type": "text",
      "value": "People & Blogs",
    },
    "channel_id": TextTableValue {
      "type": "text",
      "value": "UCoY-bu7ZpUhCOd2sJpLGQ2w",
    },
    "description": TextTableValue {
      "type": "text",
      "value": "",
    },
    "dislike_count": IntegerTableValue {
      "type": "integer",
      "value": 0,
    },
    "duration": TextTableValue {
      "type": "text",
      "value": "35 seconds",
    },
    "embeddable": TextTableValue {
      "type": "text",
      "value": "allowed",
    },
    "license": TextTableValue {
      "type": "text",
      "value": "standard YouTube",
    },
    "like_count": IntegerTableValue {
      "type": "integer",
      "value": 0,
    },
    "location_address": TextTableValue {
      "type": "text",
      "value": "",
    },
    "location_country": TextTableValue {
      "type": "text",
      "value": "",
    },
    "status": TextTableValue {
      "type": "text",
      "value": "processed",
    },
    "syndication": TextTableValue {
      "type": "text",
      "value": "all platforms",
    },
    "time_created": DateTableValue {
      "type": "date",
      "value": 1628248337000,
    },
    "time_published": DateTableValue {
      "type": "date",
      "value": null,
    },
    "time_recorded": TextTableValue {
      "type": "text",
      "value": "",
    },
    "title": TextTableValue {
      "type": "text",
      "value": "Razor review",
    },
    "video_id": TextTableValue {
      "type": "text",
      "value": "2sXKpQwAZ6Y",
    },
    "view_count": IntegerTableValue {
      "type": "integer",
      "value": 3,
    },
    "visibility": TextTableValue {
      "type": "text",
      "value": "Unlisted",
    },
  },
  Videos {
    "category": TextTableValue {
      "type": "text",
      "value": "People & Blogs",
    },
    "channel_id": TextTableValue {
      "type": "text",
      "value": "UCoY-bu7ZpUhCOd2sJpLGQ2w",
    },
    "description": TextTableValue {
      "type": "text",
      "value": "",
    },
    "dislike_count": IntegerTableValue {
      "type": "integer",
      "value": 0,
    },
    "duration": TextTableValue {
      "type": "text",
      "value": "68 seconds",
    },
    "embeddable": TextTableValue {
      "type": "text",
      "value": "allowed",
    },
    "license": TextTableValue {
      "type": "text",
      "value": "standard YouTube",
    },
    "like_count": IntegerTableValue {
      "type": "integer",
      "value": 0,
    },
    "location_address": TextTableValue {
      "type": "text",
      "value": "",
    },
    "location_country": TextTableValue {
      "type": "text",
      "value": "",
    },
    "status": TextTableValue {
      "type": "text",
      "value": "processed",
    },
    "syndication": TextTableValue {
      "type": "text",
      "value": "all platforms",
    },
    "time_created": DateTableValue {
      "type": "date",
      "value": 1628808022000,
    },
    "time_published": DateTableValue {
      "type": "date",
      "value": null,
    },
    "time_recorded": TextTableValue {
      "type": "text",
      "value": "",
    },
    "title": TextTableValue {
      "type": "text",
      "value": "2200 Willott F",
    },
    "video_id": TextTableValue {
      "type": "text",
      "value": "R_YcAF8o0zE",
    },
    "view_count": IntegerTableValue {
      "type": "integer",
      "value": 4,
    },
    "visibility": TextTableValue {
      "type": "text",
      "value": "Unlisted",
    },
  },
  Videos {
    "category": TextTableValue {
      "type": "text",
      "value": "People & Blogs",
    },
    "channel_id": TextTableValue {
      "type": "text",
      "value": "UCoY-bu7ZpUhCOd2sJpLGQ2w",
    },
    "description": TextTableValue {
      "type": "text",
      "value": "",
    },
    "dislike_count": IntegerTableValue {
      "type": "integer",
      "value": 0,
    },
    "duration": TextTableValue {
      "type": "text",
      "value": "20 seconds",
    },
    "embeddable": TextTableValue {
      "type": "text",
      "value": "allowed",
    },
    "license": TextTableValue {
      "type": "text",
      "value": "standard YouTube",
    },
    "like_count": IntegerTableValue {
      "type": "integer",
      "value": 0,
    },
    "location_address": TextTableValue {
      "type": "text",
      "value": "",
    },
    "location_country": TextTableValue {
      "type": "text",
      "value": "",
    },
    "status": TextTableValue {
      "type": "text",
      "value": "processed",
    },
    "syndication": TextTableValue {
      "type": "text",
      "value": "all platforms",
    },
    "time_created": DateTableValue {
      "type": "date",
      "value": 1628808169000,
    },
    "time_published": DateTableValue {
      "type": "date",
      "value": null,
    },
    "time_recorded": TextTableValue {
      "type": "text",
      "value": "",
    },
    "title": TextTableValue {
      "type": "text",
      "value": "2200 Willott F",
    },
    "video_id": TextTableValue {
      "type": "text",
      "value": "DlqSVM9_R9c",
    },
    "view_count": IntegerTableValue {
      "type": "integer",
      "value": 4,
    },
    "visibility": TextTableValue {
      "type": "text",
      "value": "Unlisted",
    },
  },
]
`;
